src/mlpack/core/data/load_impl.hpp

```diff
@@ -220,7 +220,9 @@ bool Load(const std::string& filename,
 
   // Now transpose the matrix, if necessary.
   if (transpose)
-    matrix = trans(matrix);
+  {
+    arma::inplace_transpose(matrix);
+  }
```

Comment:
I don't know where else we might use `inplace_transpose()`.  Do you think we should just inline it here instead of adding new files in `arma_extend/`?  I don't think this contribution will go back upstream, which is usually what `arma_extend` is for (backporting new functionality, or preparing new stuff to submit upstream).